pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -187,6 +261,280 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
+				doContinue = false
+				logger.Info("Removing deprecated rule from enableRules", "rule", rule.Name)
+				r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileDeprecatedSetting", "Removing deprecated rule: %s", rule.Name)
+				continue
+			} else {
+				newRules = append(newRules, *rule)
+			}
+		}
+		if len(newRules) != len(v1alphaTp.Spec.EnableRules) {
+			v1alphaTpCP.Spec.EnableRules = newRules
+		}
+	}
+
+	if len(v1alphaTp.Spec.DisableRules) > 0 && len(deprecatedRules) > 0 {
+		var newRules []cmpv1alpha1.RuleReferenceSpec
+		for ri := range v1alphaTp.Spec.DisableRules {
+			rule := &v1alphaTp.Spec.DisableRules[ri]
+			if _, ok := deprecatedRules[rule.Name]; ok {
+				doContinue = false
+				logger.Info("Removing deprecated rule from disableRules", "rule", rule.Name)
+				r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileDeprecatedSetting", "Removing deprecated rule: %s", rule.Name)
+				continue
+			} else {
+				newRules = append(newRules, *rule)
+			}
+		}
+		if len(newRules) != len(v1alphaTp.Spec.DisableRules) {
+			v1alphaTpCP.Spec.DisableRules = newRules
+		}
+	}
+
+	// if tp does not have any rules, variables left, and it does not extend any profile
+	// we should not update the tp, and set the state of tp to Error
+	if len(v1alphaTpCP.Spec.EnableRules) == 0 && len(v1alphaTpCP.Spec.DisableRules) == 0 && len(v1alphaTpCP.Spec.ManualRules) == 0 && v1alphaTpCP.Spec.Extends == "" {
+		errorMsg := "TailoredProfile does not have any rules left after removing deprecated rules and variables"
+		v1alphaTpCP.Status.State = cmpv1alpha1.TailoredProfileStateError
+		v1alphaTpCP.Status.ErrorMessage = errorMsg
+		doContinue = false
+		logger.Info(errorMsg)
+		r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileDeprecatedSetting", errorMsg)
+	}
+
+	if !doContinue {
+		err = r.Client.Update(context.TODO(), v1alphaTpCP)
+		logger.Info("Updating TailoredProfile with deprecated rules and variables removed")
+		if err != nil {
+			return false, err
+		}
+	}
+	return doContinue, nil
+}
+
+// getMigratedRules get list of rules and check if it has RuleLastCheckTypeChangedAnnotationKey annotation
+// if it does, add it to the map with the current check type
+func (r *ReconcileTailoredProfile) getMigratedRules(tp *cmpv1alpha1.TailoredProfile, logger logr.Logger) (map[string]string, error) {
+	// get all the rules in the namespace
+	ruleList := &cmpv1alpha1.RuleList{}
+	err := r.Client.List(context.TODO(), ruleList, &client.ListOptions{
+		Namespace: tp.GetNamespace(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// get all the rules that are migrated
+	migratedRules := make(map[string]string)
+	for ri := range ruleList.Items {
+		rule := &ruleList.Items[ri]
+		if rule.Annotations != nil {
+			if _, ok := rule.Annotations[cmpv1alpha1.RuleLastCheckTypeChangedAnnotationKey]; ok {
+				if rule.CheckType == cmpv1alpha1.CheckTypeNone {
+					logger.Info("Rule has been changed to manual check", "rule", rule.GetName())
```

Comment:
This is logging a pretty specific case, where we're informing users that not only did a rule change types, but that it went from being `Platform` or `Node` to being an empty string.

Do we also want to log the switch from `Platform` to `Node` and vice versa?

Reply:
I think the profile parser code has that logging, https://github.com/ComplianceAsCode/compliance-operator/pull/401/files#diff-607a625b7e09c78b2e136c939eee3440d30fafe1d83721b9ef6631fa789c5b7bR114